drivers/scheduler.c

```diff
@@ -294,8 +294,8 @@ scheduler_register_event(scheduler_t *s, char mode, int fd,
 	event->id       = s->uuid++;
 	event->masked   = 0;
 
-	if (!s->uuid)
-		s->uuid++;
+	if (s->uuid < 0)
+		s->uuid = 1;
```

Comment:
actually we need -ve values to signify an error condition and also there are other structs such as conn which use the same event ids. Incase if we were to change in one place and leave all others as int this will cause a valid positive event_id returned by scheduler_register_events to be treated as a -ve value since in other places its still int. On the other hand if we make all the event_ids as unsigned then there will be noway to propagate the error condition from one function to another. So I felt it was safe to loop around than play with the data types, the existing code also wanted to do the same thing except that the programmer assumed the value would loop around 0 by default.